Isaicp 2709 - Add/remove membership actions - bulk operation

sandervd opened the pull request.

sandervd committed:
Issue #ISAICP-2709: Add actions to members.

sandervd committed:
Issue #ISAICP-2709: Assure that roles are visible in member view.

sandervd committed:
Issue #ISAICP-2709: Add roles and actions to membership view.

sandervd committed:
Issue #ISAICP-2709: Fix remove roles.

sandervd committed:
Issue #ISAICP-2709: Remove member and non member from actions.

amitaibu commented:
@sandervd  Thanks, can you please open the PR against https://github.com/Gizra/og

sandervd committed:
Issue #ISAICP-2709: Add schema for bulk member role field, and update
test.

sandervd committed:
Issue #ISAICP-2709: Array to short syntax.

pfrenssen committed:
Merge remote-tracking branch 'upstream/8.x-1.x' into ISAICP-2709

pfrenssen committed:
Only allow to change member roles when the user is a group administrator or can manage members.

pfrenssen committed:
I like green.

pfrenssen committed:
Fix coding standards.

pfrenssen committed:
Don't create actions to add or remove required roles.

pfrenssen committed:
Add missing methods to OgRoleInterface.

pfrenssen committed:
Update documentation.

pfrenssen committed:
Improve variable name.

pfrenssen committed:
Complete the list of actions that can be taken on user memberships.

pfrenssen committed:
Introduce OgMembershipInterface::revokeRoleById() so it is no longer required to load a role in order to revoke it.

pfrenssen committed:
Add missing method declaration.

pfrenssen committed:
Simplify the logic used to add or revoke roles.

pfrenssen committed:
Show OgRoles in the configuration form, rather than user roles.

pfrenssen committed:
Add a type hint.

pfrenssen committed:
Move the bulk form operation checkbox into the first column of the table.

pfrenssen committed:
Fix random failure.

pfrenssen committed:
The bulk operation checkbox has moved to the first column.

pfrenssen committed:
Create actions per role name instead of per ID, to avoid duplicate actions.

pfrenssen committed:
Declare that OgRoles are dependent on their parent bundle.

This will cause them to be cleaned up automatically when the parent bundle is
deleted, we do not need to do it manually.

pfrenssen committed:
Make sure that the group configuration is updated when a group bundle is deleted.

pfrenssen committed:
Standardize variable naming.

pfrenssen committed:
Merge remote-tracking branch 'gizra/8.x-1.x' into ISAICP-2709

pfrenssen committed:
Merge branch 'remove-role-override' into ISAICP-2709

pfrenssen committed:
Test that the actions to add or remove a role are correctly created / deleted when the parent role is created / deleted.

pfrenssen committed:
Make sure the group list is updated when a group bundle is deleted, also when og_ui is not enabled.

pfrenssen committed:
Test that a bundle is no longer marked as a group after it is deleted.

pfrenssen committed:
Merge remote-tracking branch 'gizra/8.x-1.x' into ISAICP-2709

pfrenssen committed:
Test that the roles get deleted automatically if a parent group type is deleted.

pfrenssen committed:
Merge branch 'roles-depend-on-groups' into ISAICP-2709

pfrenssen committed:
Test the AddOgMembershipRole action.

pfrenssen committed:
Add tests for remaining action plugins.

pfrenssen committed:
Fix bugs uncovered by the tests.

pfrenssen committed:
Fix coding standards.

pfrenssen committed:
Merge remote-tracking branch 'gizra/8.x-1.x' into ISAICP-2709

pfrenssen committed:
Merge remote-tracking branch 'gizra/8.x-1.x' into ISAICP-2709

pfrenssen committed:
Rename the actions that affect one single role so they can be distinguished from the ones that affect multiple roles.

pfrenssen committed:
Use the correct role type.

pfrenssen committed:
Leverage OgAccess.

pfrenssen committed:
Start implementing the actions to change multiple roles at once.

pfrenssen committed:
Use the interface in type hints.

pfrenssen committed:
Add missing methods to the interface.

pfrenssen committed:
Add a method to check if a role exists in a membership.

pfrenssen committed:
Add a static method that returns all roles that are associated with a group type.

pfrenssen committed:
Factor out the code that checks if a role is a required default role so it can be reused.

pfrenssen committed:
Add actions that allow to add or remove multiple roles using bulk operations.

pfrenssen committed:
Add new actions to config schema.

pfrenssen committed:
Remove unused use statements.

pfrenssen committed:
Add documentation.

pfrenssen committed:
Merge branch 'enforce-correct-roles' into ISAICP-2709

pfrenssen committed:
Hide the actions to add/remove single roles from the bulk operations by default.

pfrenssen committed:
Rename the test classes dealing with changes to a single membership role, so they can be distinguished from the ones that deal with multiple roles.

pfrenssen committed:
Test the actions to add/remove multiple roles.

pfrenssen committed:
Dramatically speed up test runs by executing all test cases in a single test.

pfrenssen committed:
Make the adding of roles more robust by first checking if the roles are valid.

pfrenssen committed:
Update documentation.

pfrenssen committed:
Fix coding standards.

pfrenssen committed:
Add a development hint in a todo.

pfrenssen committed:
Allow removal of roles when the parent entity is being removed.

pfrenssen committed:
Fix coding standards.

pfrenssen committed:
Merge remote-tracking branch 'gizra/8.x-1.x' into ISAICP-2709

pfrenssen committed:
The single role action plugins are not depending on particular OG Roles.

They can actually depend on a range of different roles for different group
types, with the same role name.

The cleaning up of these action plugins is handled in og_og_role_delete().

pfrenssen committed:
Merge branch 'delete-group-type' into ISAICP-2709

pfrenssen committed:
Merge remote-tracking branch 'origin/8.x-1.x' into ISAICP-2709

pfrenssen committed:
Merge branch 'use-interface' into ISAICP-2709

pfrenssen committed:
Merge branch 'fix-random-failure' into ISAICP-2709

pfrenssen committed:
Fix documentation.

pfrenssen committed:
Merge remote-tracking branch 'gizra/8.x-1.x' into ISAICP-2709

pfrenssen committed:
Merge remote-tracking branch 'gizra/delete-group-type' into ISAICP-2709

pfrenssen committed:
Merge remote-tracking branch 'origin/8.x-1.x' into ISAICP-2709

pfrenssen committed:
Remove duplicate function, was left behind during merge.

pfrenssen committed:
Merge remote-tracking branch 'gizra/8.x-1.x' into ISAICP-2709

pfrenssen committed:
Merge remote-tracking branch 'gizra/8.x-1.x' into ISAICP-2709

pfrenssen committed:
Merge branch 'load-roles-by-group-type' into ISAICP-2709

pfrenssen committed:
Merge branch 'is-default-role' into ISAICP-2709

pfrenssen committed:
Merge branch 'is-default-role' into ISAICP-2709

pfrenssen committed:
Rename isDefaultRole() to isRequired().

pfrenssen committed:
Merge remote-tracking branch 'github/load-roles-by-group-type' into ISAICP-2709

pfrenssen committed:
Merge remote-tracking branch 'github/is-default-role' into ISAICP-2709

pfrenssen committed:
Merge remote-tracking branch 'gizra/8.x-1.x' into ISAICP-2709

pfrenssen committed:
Leverage the new OgRoleInterface::isRequired() method.

pfrenssen committed:
Remove unused use statement.

claudiu-cristea committed:
Single role actions labels should depend on the group context.

pfrenssen committed:
Merge pull request #2 from ec-europa/ISAICP-3577

Single role actions labels should depend on the group context

pfrenssen committed:
Update call to method that was recently renamed.

pfrenssen committed:
Provide a base class containing shared code for action plugins that change membership state.

pfrenssen committed:
Provide an action plugin that allows to change a membership state to 'pending'.

pfrenssen committed:
Use uppercase for boolean values in return statement documentation.

pfrenssen committed:
Add a method to check if a membership belongs to the group owner.

pfrenssen committed:
Remove check on undefined variable.

pfrenssen committed:
The group owner's state should never be changed.
